### PR TITLE
Use SceneId enum

### PR DESCRIPTION
**Pull request by GnomedDev**
Renames `System::RKSceneId` to `System::SceneId` and uses it more.

### CODE REVIEW

File: payload/egg/core/eggSceneManager.cc [hunk 1]
```diff
@@ -100,11 +100,11 @@ void SceneManager::reinitCurrentScene() {
     }
 }
 
-void SceneManager::createScene(s32 sceneId, Scene *parent) {
+void SceneManager::createScene(EGG::SceneId sceneId, Scene &parent) {
```
Comment (vabold) [inline payload/egg/core/eggSceneManager.cc:103]:
```suggestion
void SceneManager::createScene(EGG::SceneId sceneId, Scene *parent) {
```
The function explicitly checks if `parent` is a nullptr.

File: payload/egg/core/eggScene.hh [hunk 2]
```diff
@@ -4,6 +4,21 @@
 
 namespace EGG {
 
+enum class SceneId : u32 {
```
Comment (vabold) [inline payload/egg/core/eggScene.hh:7]:
```suggestion
enum class SceneId {
```

File: payload/egg/core/eggSceneManager.cc [hunk 3]
```diff
@@ -87,10 +87,10 @@ void SceneManager::reinitCurrentScene() {
...
     // graphical bugs. Opted to reload the entire track as a simple fix.
     if (rc->raceScenario().mirror && (setting != SP::ClientSettings::TAClass::Mirror)) {
-        Scene *parent = m_currScene->getParent();
-        u32 sceneID = m_currScene->getSceneID();
-        destroyScene(m_currScene);
-        createScene(sceneID, parent);
+        auto parent = m_currScene->getParent();
+        auto sceneID = m_currScene->id();
+        destroyScene(*m_currScene);
+        createScene(sceneID, *parent);
```
Comment (vabold) [inline payload/egg/core/eggSceneManager.cc:93]:
```suggestion
        auto *parent = m_currScene->getParent();
        auto sceneID = m_currScene->id();
        destroyScene(*m_currScene);
        createScene(sceneID, parent);
```

File: payload/egg/core/eggSceneManager.hh [hunk 4]
```diff
@@ -6,14 +6,17 @@ namespace EGG {
...
+    SceneManager();
+    virtual ~SceneManager();
+
     void REPLACED(reinitCurrentScene)();
     void REPLACE reinitCurrentScene();
 
-    void REPLACED(createScene)(s32 sceneId, Scene *parent);
-    void REPLACE createScene(s32 sceneId, Scene *parent);
+    void REPLACED(createScene)(EGG::SceneId sceneId, Scene &parent);
+    void REPLACE createScene(EGG::SceneId sceneId, Scene &parent);
```
Comment (vabold) [inline payload/egg/core/eggSceneManager.hh:16]:
```suggestion
    void REPLACED(createScene)(EGG::SceneId sceneId, Scene *parent);
    void REPLACE createScene(EGG::SceneId sceneId, Scene *parent);
```

**Review by vabold — CHANGES_REQUESTED** [review]

**Comment by vabold** [comment]
It's likely that we do not want to move the game-specific scene IDs out of `host_system` and into `EGG` as the library is not game-dependent.

File: payload/game/ui/Section.hh [hunk 5]
```diff
@@ -3,6 +3,7 @@
 #include "game/ui/Page.hh"
 #include "game/ui/SectionId.hh"
 
+#include <game/host_system/Scene.hh>
 #include <nw4r/lyt/lyt_drawInfo.hh>
```
Comment (vabold) [inline payload/game/ui/Section.hh:7]:
```suggestion
#include "game/host_system/Scene.hh"

#include <nw4r/lyt/lyt_drawInfo.hh>
```

**Review by vabold — CHANGES_REQUESTED** [review]

**Comment by vabold** [comment]
Nice work!

File: payload/game/ui/Section.hh (hunk 5 above)
Comment (GnomedDev) [reply payload/game/ui/Section.hh]: Done

**Review by vabold — APPROVED** [review]

**Review by stblr — APPROVED** [review]

**Pull request merged**